microprofile/WFLY-17137-microprofile-jakarta-ee-10.adoc

```diff
@@ -0,0 +1,200 @@
+:author:            Kabir Khan
+:email:             kkhan@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+This issue is about upgrading to most parts of MicroProfile Platform 6, which is aligned with Jakarta EE 10. 'most' is used since a decision has been made to not include the new MicroProfile Metrics version, as the inclusion of Micrometer tracked by https://issues.redhat.com/browse/WFLY-14947[WFLY-14947] / https://issues.redhat.com/browse/WFLY-17144[WFLY-17144] provides our observability strategy.
+
+The below table shows the versions of each specification in MicroProfile Platform 5, the current versions included in WildFly, and the planned version for MicroProfile Platform 6. Only the minor versions are included, we will use the latest micro versions of each at the time of release. It also contains a column indicating whether something changes the functionality provided by a specification. Note that we are just pointing out the minor versions of specifications here. There may be micro updates, as pointed out in the <<hard-requirements, Hard Requirements>> section later.
+
+[cols="1,1,1,1,1"]
+|===
+|Spec | MP 5 Version | Current WildFly Version | MP 6 Version | Updated Functionality (y/n)
+
+|Config
+|3.0
+|3.0
+|3.0
+|n
+
+|Fault Tolerance
+|4.0
+|4.0
+|4.0
+|n
+
+|Health
+|4.0
+|4.0
+|4.0
+|n
+
+|JWT
+|2.0
+|2.0
+|2.1
+|y
+
+|+++<s>Metrics</s>+++ +
+Will not include
+|+++<s>4.0</s>+++
+|+++<s>4.0</s>+++
+|+++<s>5.0</s>+++
+|+++<s>y</s>+++
+
+|OpenAPI
+|3.0
+|3.0
+|3.1
+|y (trivial)
+
+|OpenTracing +
+Dropped in MP 6
+|3.0
+|3.0
+|-
+|-
+
+|Rest Client
+|3.0
+|3.0
+|3.0
+|n
+
+|Telemetry +
+New in MP 6
+|-
+|-
+|1.0
+|y
+
+|===
+
+There will be a separate RFEs for:
+
+* Removal of MicroProfile Metrics (https://issues.redhat.com/browse/WFLY-17138[WFLY-17138]). The functional replacement is Micrometer, listed below.
+* Removal of MicroProfile OpenTracing (https://issues.redhat.com/browse/WFLY-17510/[WFLY-17510]). The functional replacement is Micrometer, listed below.
+* Addition of MicroProfile Telemetry (https://issues.redhat.com/browse/WFLY-17156[WFLY-17156])
+* Addition of Micrometer (https://issues.redhat.com/browse/WFLY-14947[WFLY-14947])
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-17137[WFLY-17137] - Provide Support for MicroProfile.Next version that Aligns with Jakarta EE 10
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1861[EAP7-1861] - Provide Support for MicroProfile.Next version that Aligns with Jakarta EE 10
+* https://issues.redhat.com/browse/WFLY-17138[WFLY-17138] - Convert MP Metrics subsystem to Legacy Mode
+* https://issues.redhat.com/browse/WFLY-17156[WFLY-17156] - Implement Support for MicroProfile Telemetry
+* https://issues.redhat.com/browse/WFLY-17510/[WFLY-17510] - Removal of MicroProfile OpenTracing
+* https://issues.redhat.com/browse/WFLY-14947[WFLY-14947] - Addition of Micrometer
+*
```

Comment:
Shouldn't https://issues.redhat.com/browse/WFLY-17144 be listed in here too?

Reply:
Added